src/Nethermind/Nethermind.JsonRpc/Data/ForkActivationParameter.cs

```diff
@@ -0,0 +1,190 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using Nethermind.Core.Specs;
+using Nethermind.Specs.Forks;
+
+namespace Nethermind.JsonRpc.Data;
+
+/// <summary>
+/// Parameter for specifying fork specification overrides for trace operations.
+/// Allows tracing blocks with different protocol rules than their original fork.
+/// 
+/// Examples:
+/// - {"forkName": "osaka"} - Use Osaka fork rules
+/// - {"activationBlock": 1000000} - Override activation block
+/// - {"forkName": "prague", "activationBlock": 0} - Use Prague from genesis
+/// </summary>
+public class ForkActivationParameter
+{
+    private static readonly Dictionary<string, Func<IReleaseSpec>> ForkRegistry = new(StringComparer.OrdinalIgnoreCase)
+    {
+        ["frontier"] = () => Frontier.Instance,
```

Comment:
could use nameof for guaranteed consistency
```suggestion
        [nameof(Frontier)] = () => Frontier.Instance,
```